lib/src/components/NxDrawer/NxDrawer.scss

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use '../../scss-shared/nx-container-helpers';
+ @use '../../scss-shared/nx-text-helpers';
+ 
+$-box-shadow-offset: 52px;
+
+.nx-drawer {
+  // Undo styles that chrome applies to <dialog> elements
+  border: initial;
+  height: initial;
+  left: initial;
+  margin: initial;
+  max-height: initial;
+  max-width: initial;
+  padding: initial;
+  width: initial;
+
+  // Prevent brief flash of scrollbar as tooltips disappear
+  overflow: hidden;
+
+  &::backdrop {
+    // Chrome tries to add its own subtle mask here which doubles up with our own
+    display: none;
+    pointer-events: auto;
+  }
+
+  background: transparent;
+  box-sizing: border-box;
+  height: 100%;
+  padding-left: $-box-shadow-offset;
+  position: fixed;
+  right: 0;
+  top: 0;
+  transform: translateX(100%);
+  width: 520px + $-box-shadow-offset;
+  z-index: 100;
+
+  &--with-global-header {
```

Comment:
Rather than this, would it be acceptable to define `NxDrawer` as a page-level component along the same lines as `NxPageMain`? E.g. a component that may only be used as a direct child of `.nx-page` or `.nx-page-content`?  Then it could be managed by the `.nx-page` grid which would allow easy control of how it interacts with things like headers.

Reply:
> Rather than this, would it be acceptable to define `NxDrawer` as a page-level component along the same lines as `NxPageMain`? E.g. a component that may only be used as a direct child of `.nx-page` or `.nx-page-content`? Then it could be managed by the `.nx-page` grid which would allow easy control of how it interacts with things like headers.

This was fixed as part of these commits:
3f200126b91f0977a570a531d67c17d92ed1f323
extra commits to fix the animations:
34abea79f4e3011eb4d06cf50ad039a187b341e4
59bd05030b537a579817f5d9262f679d6b0f4565